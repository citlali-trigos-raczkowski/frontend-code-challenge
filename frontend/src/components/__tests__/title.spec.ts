import { shallowMount, createLocalVue } from "@vue/test-utils";
import Title from "../title.vue";
import VueRouter from "vue-router";
import { TitleMockData } from "../../../tests/mocks/mock-props";

const localVue = createLocalVue();
localVue.use(VueRouter);

describe("component/title.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(Title, {
      propsData: TitleMockData,
      localVue,
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("prop title is displayed", () => {
    const wrapper = shallowMount(Title, {
      propsData: TitleMockData,
      localVue,
    });
    expect(wrapper.text()).toContain("");
  });

  it("includes link back home", () => {
    const wrapper = shallowMount(Title, {
      propsData: TitleMockData,
      localVue,
    });
    const aWrapper = wrapper.find("router-link-stub");
    expect(aWrapper.attributes().to).toBe("/");
  });
});
