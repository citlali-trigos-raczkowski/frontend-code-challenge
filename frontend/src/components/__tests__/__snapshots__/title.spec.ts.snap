// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component/title.vue renders correctly 1`] = `
<div
  id="title-box"
  pagetitle="Page Title Sample"
>
  <router-link-stub
    ariacurrentvalue="page"
    event="click"
    id="title"
    tag="a"
    to="/"
  >
    <h1
      bold=""
    >
      Who's That Pokemon?
    </h1>
  </router-link-stub>
</div>
`;
